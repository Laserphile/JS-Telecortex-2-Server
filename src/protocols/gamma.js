/**
 * Build a gamma table for a particular value
 * Stolen from https://github.com/ajfisher/node-pixel/blob/9f956c26cd5dbe16f6066eb659cf08350c8b3fea/lib/pixel.js#L401
 */
export const createGammaTable = (steps, gamma) =>
  Array.from({ length: steps }).map((_, i) => Math.floor((i / 255.0) ** gamma * 255 + 0.5));
