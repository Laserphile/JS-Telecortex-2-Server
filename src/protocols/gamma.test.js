import { createGammaTable } from './gamma'

describe('createGammaTable', () => {
  it('works', ()=>{
    expect(createGammaTable(3, 1.5)).toMatchSnapshot();
  })
});
