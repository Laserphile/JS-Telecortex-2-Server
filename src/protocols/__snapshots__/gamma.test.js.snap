// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createGammaTable works 1`] = `
Array [
  0,
  0,
  0,
]
`;
